### PR TITLE
backport changes on ThresholdedRandomCutForest integration

### PR DESCRIPTION
**Pull request by kaituo**
PR in the main branch: https://github.com/opensearch-project/anomaly-detection/pull/238
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kaituo** [comment]
Test failed with flaky test: https://github.com/opensearch-project/anomaly-detection/pull/239/checks?check_run_id=3724255448.  Rerun workflow.
 65 tests completed, 1 failed, 1 skipped
Tests with failures:
 - org.opensearch.ad.rest.AnomalyDetectorRestApiIT.testUpdateAnomalyDetectorWithRunningAdJob

**Review by ylwu-amzn — APPROVED** [review]

File: src/main/java/org/opensearch/ad/ml/ModelManager.java [hunk 1]
```diff
@@ -469,8 +469,7 @@ private void trainModelForStep(
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.REAL_TIME_BOUNDING_BOX_CACHE_RATIO)
-            // for external shingling, rcf does not recognize shingle
-            .shingleSize(1)
+            .shingleSize(detector.getShingleSize())
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/ml/ModelManager.java:472]: good

File: src/main/java/org/opensearch/ad/ml/ModelManager.java [hunk 2]
```diff
@@ -559,8 +558,7 @@ public void maintenance(ActionListener<Void> listener) {
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
-            // for external shingling, rcf does not recognize shingle
-            .shingleSize(1)
+            .shingleSize(shingleSize)
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/ml/ModelManager.java:561]: good

File: src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java [hunk 3]
```diff
@@ -92,11 +92,7 @@ protected ADBatchTaskCache(ADTask adTask) {
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
-            // for external shingling, rcf does not recognize shingle. Thus, shingle size
-            // is 1 here.
-            // shingle in detector config and shingle size here are different things.
-            // shingle size in detector config impacts dimensions.
-            .shingleSize(1)
+            .shingleSize(shingleSize)
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java:95]: good

**Review by sudiptoguha — APPROVED** [review]

**Pull request merged**